bookkeeper-server/src/test/java/org/apache/bookkeeper/client/api/BKExceptionTest.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2017 The Apache Software Foundation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.bookkeeper.client.api;
+
+import static org.apache.bookkeeper.client.api.BKException.Code.UnexpectedConditionException;
+import static org.hamcrest.Matchers.equalTo;
```

Comment:
any reason why not use junit Asserts? does it not meet your requirements?

Reply:
@sijie 
I saw that in another commit we started to use *assertThat*, recently I started to use it even for "equals", usually error messages generated automatically are more readable (like the one for greaterThan)
I have changed this, now using basic assertEquals/assertNotEquals